runtime/parachains/src/dmp.rs

```diff
@@ -228,7 +241,17 @@ impl<T: Config> Pallet<T> {
 			} else {
 				*q = q.split_off(processed_downward_messages);
 			}
+			q.len()
 		});
+
+		let config = configuration::ActiveConfig::<T>::get();
+		if q_len <=
+			(MAX_POSSIBLE_ALLOCATION
```

Comment:
Is this a best effort check or what does it ensure?

Reply:
Based on this comment (https://github.com/paritytech/srlabs_findings/issues/203#issuecomment-1477778761) we calculated the limit for increasing the fees to be: 
Max_allocation (32Mb) / Max_message_size (currently 50 kb) = 640 messages. So we currently accept 640 max sized messages in the queue before we reach the max_alloc. Our best effort approach was dividing 640 in half for our limit. However if you have a better approach please let me know!

Reply:
I'd probably extract this equation into a let binding and call it something meaningful, like `delivery_fee_limit` or something.